library/src/main/kotlin/io/intrepid/bleidiom/BleIdiomImpl.kt

```diff
@@ -190,39 +193,39 @@ internal class BleCharValueDelegate<Val : Any>() : BleCharHandlerDSL<Val> {
                         else it.setupNotification(charUUID)
                     }
                     setupObserver()
-                            .flatMap { notificationSetup -> notificationSetup }
-                            .toRx2()
-                            .map { byteArray -> transform(byteArray) }
-                            .doOnNext { backingField.currentValue = it }
+                        .flatMap { notificationSetup -> notificationSetup }
+
+                        .map { byteArray -> transform(byteArray) }
+                        .doOnNext { backingField.currentValue = it }
                 }.takeUntil(service!!.killedConnectionObs)
             } ?: Observable.error(IllegalStateException("observeAction is null"))
         }
 
         backingField.writeAction = { value ->
             val mtu = service?.mtuSize ?: MIN_MTU_SIZE
             letMany(uuid, service?.sharedConnection, toByteArray) { charUUID, connection, transform ->
-                connection.firstOrError().flatMapTry {
+                connection.firstOrError().flatMapTry { rxBleConnection : RxBleConnection ->
                     val bytes = transform(value)
-                    val writeObs = if (bytes.size > mtu) {
+                    val writeObs : Single<ByteArray> = if (bytes.size > mtu) {
                         val (batchSize, batchBytes) = toBatchInfo(value, bytes)
                         if (batchSize in 1..mtu) {
-                            it.createNewLongWriteBuilder()
-                                    .setCharacteristicUuid(charUUID)
-                                    .setMaxBatchSize(batchSize)
-                                    .setBytes(batchBytes)
-                                    .build()
+                            Single.fromObservable<ByteArray> {
```

Comment:
Remove the `<ByteArray>`. The type can be inferred automatically.

Or remove the `Single.fromObservable` entirely and add `singleOrError()` instead:
```
                            rxBleConnection.createNewLongWriteBuilder()
                                    .setCharacteristicUuid(charUUID)
                                    .setMaxBatchSize(batchSize)
                                    .setBytes(batchBytes)
                                    .build()
                                    .singleOrError()
```